### ISSUE TITLE
flutter 1.0: canvas.drawArc doesn't work as expected when useCenter is false, but if useCenter is true, It's work

### ISSUE THREAD

**Issue by bowen919446264** [flutter/flutter#25069]
## Steps to Reproduce

1. code :
```dart
   rect = Rect.fromCircle(center: center, radius: radius);
    angle = 160.0 * (-1);
    double startAngle = -math.pi / 2;
    double sweepAngle = 2 * math.pi * (angle / 360);
    print(
        'angle === $angle --- $startAngle --- $sweepAngle --- ${rect.toString()} --- $progressPaint');
    canvas.drawArc(rect, startAngle, sweepAngle, false, progressPaint);
```
 It doesn't work
2. 
```dart
canvas.drawArc(rect, startAngle, sweepAngle, true, progressPaint);
```
It work, Why?

**Comment by zoechi** [comment]
What does "doesn't work" and "work" mean exactly. 
What is the actual and expected behavior?

Please add the output of `flutter doctor -v`.

**Comment by bowen919446264** [comment]
I want to draw an arc, not including the center of the circle.  the code no work
canvas.drawArc(rect, startAngle, sweepAngle, false, progressPaint);
I try set useCenter is true; It's work, But it is not what I expected

**Comment by bowen919446264** [comment]
[√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Version 10.0.17134.407], locale zh-CN)
    • Flutter version 1.0.0 at E:\fluter\flutter_windows_v0.9.4-beta\flutter
    • Framework revision 5391447fae (7 days ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at E:\android_config\android-sdk-windows
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-P, build-tools 27.0.3
    • ANDROID_HOME = E:\android_config\android-sdk-windows
    • Java binary at: C:\Program Files\Java\jdk1.8.0_144\bin\java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
    • All Android licenses accepted.

[!] Android Studio (version 2.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[!] IntelliJ IDEA Community Edition (version 14)
    • IntelliJ at C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 14.1.4
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins
    X This install is older than the minimum recommended version of 2017.1.0.

[√] IntelliJ IDEA Ultimate Edition (version 2017.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2017.2.1
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.3544.34

[!] VS Code, 64-bit edition (version 1.21.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (1 available)
    • vivo Y67A • SK4HNB9L5LWGQKSO • android-arm64 • Android 6.0 (API 23)

! Doctor found issues in 3 categories.

**Comment by zoechi** [comment]
So you don't get an error message.
Could you post screenshot to make it more clear what the actual vs expected behavior is?

**Comment by bowen919446264** [comment]
`Rect rect = Rect.fromCircle(center: center, radius: radius);
    angle = 160.0 * (-1);
    double startAngle = -math.pi / 2;
    double sweepAngle = 2 * math.pi * (angle / 360);
    print(
        'angle === $angle --- $startAngle --- $sweepAngle --- ${rect.toString()} --- $progressPaint');
    //log: -160.0 --- -1.5707963267948966 --- -2.792526803190927 --- Rect.fromLTRB(-9.5, -9.5, 34.5, 34.5) --- Paint(PaintingStyle.stroke 4.0 StrokeCap.round StrokeJoin.miter up to 0.0; Color(0xfff44336))
    canvas.drawArc(rect, startAngle, sweepAngle, false, progressPaint);`
[imag_bug](https://github.com/bowen919446264/demowelcome/blob/master/bug2.png)

but using code `canvas.drawArc(rect, startAngle, sweepAngle, true, progressPaint);`

[img_work](https://github.com/bowen919446264/demowelcome/blob/master/bug1.png)

**Comment by bowen919446264** [comment]
`Rect rect = Rect.fromCircle(center: center, radius: radius);
    angle = 160.0 * (-1);
    double startAngle = -math.pi / 2;
    double sweepAngle = 2 * math.pi * (angle / 360);
    print(
        'angle === $angle --- $startAngle --- $sweepAngle --- ${rect.toString()} --- $progressPaint');
    //log: -160.0 --- -1.5707963267948966 --- -2.792526803190927 --- Rect.fromLTRB(-9.5, -9.5, 34.5, 34.5) --- Paint(PaintingStyle.stroke 4.0 StrokeCap.round StrokeJoin.miter up to 0.0; Color(0xfff44336))
    canvas.drawArc(rect, startAngle, sweepAngle, false, progressPaint);`

[no word](https://github.com/bowen919446264/demowelcome/blob/master/bug2.png)

but, canvas.drawArc(rect, startAngle, sweepAngle, true, progressPaint);

[has use center](https://github.com/bowen919446264/demowelcome/blob/master/bug1.png)

**Comment by zoechi** [comment]
I assume your last comment is a duplicate posted by mistake?

**Comment by bowen919446264** [comment]
yes.

**Comment by zoechi** [comment]
I deleted it to avoid confusion.

**Comment by StarJiao** [comment]
met the same situation as @zywj ,it is so nice if you will pay attention to this. @zoechi 
more detail is :
The drawArc method didn't work if drawCircle also exists and drawCircle use PaintingStyle.fill
my code is

```
import 'dart:math';

import 'package:flutter/material.dart';

class CircleEndsProgressBar extends CustomPainter {
  CircleEndsProgressBar({
    this.percent,
    this.arcColor,
    this.innerColor,
  });

  final double percent;
  Offset center = Offset(100.0, 100.0);
  double radius = 80.0;
  double thickness = 10.0;
  Color innerColor = Colors.red;
  Color arcColor = Colors.green;

  @override
  void paint(Canvas canvas, Size size) {
    var circlePaint = new Paint()
      ..color = innerColor
      ..strokeWidth = 100
      ..style = PaintingStyle.fill;
    canvas.drawCircle(center, 10, circlePaint);
    var broaderPaint = new Paint()
      ..color = arcColor
      ..strokeWidth = thickness
      ..style = PaintingStyle.stroke
      ..strokeCap = StrokeCap.round;
    canvas.drawArc(Rect.fromCircle(center: center, radius: radius + 10), -0.5 * pi, 2 * percent * pi, false, broaderPaint);
  }

  @override
  bool shouldRepaint(CircleEndsProgressBar oldDelegate) {
    return oldDelegate.percent != percent;
  }
}
```

flutter doctor output:

```
 Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.0.0, on Mac OS X 10.14 18A391, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.30.0)
[✓] Connected device (3 available)

• No issues found!
```

**Comment by kangwang1988** [comment]
@liyuqian

**Comment by Androbin** [comment]
Same issue here. If `useCenter` is `false`, then various other elements aren't drawn at all.

**Comment by liyuqian** [comment]
I've confirmed the issue in v1.0.0-v1.1.2. It's fixed in v1.1.3 so please update Flutter to fix the issue :smile:

**Comment by zoechi** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
Thanks for your contribution.
